src/feeds/WbtcPriceFeed.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.8.20;
+
+interface IChainlinkFeed {
+    function aggregator() external view returns(address aggregator);
+    function decimals() external view returns (uint8 decimals);
+    function latestRoundData() external view returns (uint80 roundId, int256 crvUsdPrice, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound);
+}
+interface IAggregator {
+    function maxAnswer() external view returns(int192);
+    function minAnswer() external view returns(int192);
+}
+interface ICurvePool {
+    function price_oracle(uint k) external view returns (uint256);
+}
+
+
+contract WbtcPriceFeed {
+    
+    IChainlinkFeed public btcToUsd = IChainlinkFeed(0xF4030086522a5bEEa4988F8cA5B36dbC97BeE88c);
+    IChainlinkFeed public wbtcToBtc = IChainlinkFeed(0xfdFD9C85aD200c506Cf9e21F1FD8dd01932FBB23);
+    ICurvePool public tricrypto = ICurvePool(0xf5f5B97624542D72A9E06f04804Bf81baA15e2B4);
+
+    function decimals() external view returns (uint8){
+        return 8;
+    }
+
+    /**
+     * @notice Retrieves the latest round data for the wBTC token price feed
+     * @dev This function calculates the wBTC price in USD by combining the BTC to USD price from a Chainlink oracle and the wBTC to BTC ratio from another Chainlink oracle
+     * @return roundId The round ID of the Chainlink price feed for the feed with the lowest updatedAt feed
+     * @return wbtcUsdPrice The latest wBTC price in USD computed from the Wbtc/BTC and BTC/USD feeds
+     * @return startedAt The timestamp when the latest round of Chainlink price feed started of the lowest last updatedAt feed
+     * @return updatedAt The lowest timestamp when either of the latest round of Chainlink price feed was updated
+     * @return answeredInRound The round ID in which the answer was computed of the lowest updatedAt feed
+     */
+    function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80){
+        (uint80 wbtcBtcRoundId,int256 wbtcBtcPrice,uint wbtcBtcStartedAt,uint wbtcBtcUpdatedAt,uint80 wbtcBtcAnsweredInRound) = wbtcToBtc.latestRoundData();
+        (uint80 btcUsdRoundId,int256 btcUsdPrice,uint btcUsdStartedAt,uint btcUsdUpdatedAt,uint80 btcUsdAnsweredInRound) = btcToUsd.latestRoundData();
+        int wbtcUsdPrice = btcUsdPrice * 10**8 / wbtcBtcPrice;
+        if(isPriceOutOfBounds(wbtcBtcPrice, wbtcToBtc)){
+            wbtcUsdPrice = wbtcToUsdFallbackOracle();
```

Comment:
this is technically wbtcUsdtPrice, which means it could be affected by a USDT depeg.

Reply:
Would it make sense to have some indicator in the return data to specify if the fallback oracle was used?

Reply:
It would make sense if our current oracle supported using this information. We're planning an update that'll introduce fallback feeds in the oracle itself, with the individual price feeds just returning whether the returned price is within safety parameters (with Chainlink being a staleness check, min and max check).

Reply:
> this is technically wbtcUsdtPrice, which means it could be affected by a USDT depeg.

It's a good point. Will incorporate a chainlink feed that reads from USDT/USD in case of using the fallback.

Reply:
> this is technically wbtcUsdtPrice, which means it could be affected by a USDT depeg.

Added a conversation from USDT to USD price in the fallback oracle in [75c5735](https://github.com/InverseFinance/FiRM/pull/57/commits/75c5735b380c9c831bdd2123afdfcf041fc1d45e). This makes the fallback reliant on Chainlink, which I'm not entirely sure is a great idea, but considering this specific fallback is to guard against a WBTC depeg event, I think it's fine for the specific feed.

I don't do any checks of the soundness of the chainlink prices returned in the fallback, as WBTC reaching max or min price is already an extreme event.